kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java

```diff
@@ -31,28 +42,25 @@ public CacheSlotMetadata(
     this.cacheSlotState = cacheSlotState;
     this.replicaId = replicaId;
     this.updatedTimeEpochMs = updatedTimeEpochMs;
+    this.supportedIndexTypes = supportedIndexTypes;
   }
 
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     if (!super.equals(o)) return false;
-
     CacheSlotMetadata that = (CacheSlotMetadata) o;
-
-    if (!cacheSlotState.equals(that.cacheSlotState)) return false;
-    if (updatedTimeEpochMs != that.updatedTimeEpochMs) return false;
-    return replicaId.equals(that.replicaId);
+    return updatedTimeEpochMs == that.updatedTimeEpochMs
+        && cacheSlotState == that.cacheSlotState
+        && Objects.equal(replicaId, that.replicaId)
+        && Objects.equal(supportedIndexTypes, that.supportedIndexTypes);
   }
 
   @Override
   public int hashCode() {
-    int result = super.hashCode();
-    result = 31 * result + cacheSlotState.hashCode();
-    result = 31 * result + replicaId.hashCode();
-    result = 31 * result + (int) (updatedTimeEpochMs ^ (updatedTimeEpochMs >>> 32));
-    return result;
+    return Objects.hashCode(
```

Comment:
Wrong default in new IDE. Confused this with `Object.hash` which I think we should be moving to in the future for these classes instead of IntelliJ default.